source/base/Profiler.ooc

```diff
@@ -53,15 +53,15 @@ Profiler: class {
 	benchmark: static func (task: Func, timeoutMilliseconds := 1000.0) -> Double {
 		numberOfIterations := 0
 		timer := WallTimer new()
-		totalTime := 0.0
-		while (totalTime < timeoutMilliseconds * 1000.0) {
+		totalTime := TimeSpan new(0)
+		while (totalTime toMilliseconds() < timeoutMilliseconds) {
 			numberOfIterations += 1
 			timer start()
 			task()
 			totalTime += timer stop()
 		}
 		timer free()
-		totalTime / (numberOfIterations * 1000)
+		totalTime toMilliseconds() as Double / (numberOfIterations * 1000)
```

Comment:
I think it should work the way you suggest. There used to be problems with ooc implicit casts, but it should work now. I will try it.